torch/fx/passes/split_utils.py

```diff
@@ -310,3 +316,192 @@ def remap_func(x):
         return result_gm, orig_to_split_fqn_mapping
 
     return result_gm
+
+
+@compatibility(is_backward_compatible=False)
+def move_non_tensor_nodes_on_boundary(subgraphs) -> None:
+    """
+    Move non-tensor nodes on the boundary between subgraphs.
+
+    For each subgraph:
+
+    1. Find nodes whose type is not tensor and any of its children is in another
+       subgraph, put them in a queue for next step
+
+    2. Do a BFS on those nodes, for each node, let's say node X and it is in subgraph A:
```

Comment:
I'm thinking you're referring to `can_move_node_and_dependencies` here.

```suggestion
    2. Do a DFS on those nodes, for each node, let's say node X and it is in subgraph A:
```